core/src/main/java/org/apache/iceberg/MetadataLogTable.java

```diff
@@ -0,0 +1,116 @@
+ */
+
+package org.apache.iceberg;
+
+import java.util.List;
+import org.apache.iceberg.io.CloseableIterable;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.types.Types;
+import org.apache.iceberg.util.SnapshotUtil;
+
+public class MetadataLogTable extends BaseMetadataTable {
+
+  private static final Schema METADATA_LOG_SCHEMA = new Schema(
+      Types.NestedField.required(1, "timestamp_millis", Types.LongType.get()),
+      Types.NestedField.required(2, "file", Types.StringType.get()),
+      Types.NestedField.optional(3, "latest_snapshot_id", Types.LongType.get()),
+      Types.NestedField.optional(4, "latest_schema_id", Types.IntegerType.get()),
+      Types.NestedField.optional(5, "latest_sequence_number", Types.LongType.get())
+  );
+
+  MetadataLogTable(TableOperations ops, Table table) {
+    this(ops, table, table.name() + ".metadata_log");
+  }
+
+  MetadataLogTable(TableOperations ops, Table table, String name) {
+    super(ops, table, name);
+  }
+
+  @Override
+  MetadataTableType metadataTableType() {
+    return MetadataTableType.METADATA_LOG;
+  }
+
+  @Override
+  public TableScan newScan() {
+    return new MetadataLogScan(operations(), table());
+  }
+
+  @Override
+  public Schema schema() {
+    return METADATA_LOG_SCHEMA;
+  }
+
+  private DataTask task(TableScan scan) {
+    TableOperations ops = operations();
+    List<TableMetadata.MetadataLogEntry> metadataLogEntries =
+        Lists.newArrayList(ops.current().previousFiles().listIterator());
+    metadataLogEntries.add(new TableMetadata.MetadataLogEntry(ops.current().lastUpdatedMillis(),
+        ops.current().metadataFileLocation()));
+    return StaticDataTask.of(
+        ops.io().newInputFile(ops.current().metadataFileLocation()),
+        schema(),
+        scan.schema(),
+        metadataLogEntries,
+        metadataLogEntry -> MetadataLogTable.metadataLogTableToRow(metadataLogEntry, table())
+    );
+  }
+
+  private class MetadataLogScan extends StaticTableScan {
+    MetadataLogScan(TableOperations ops, Table table) {
+      super(ops, table, METADATA_LOG_SCHEMA, MetadataTableType.METADATA_LOG, MetadataLogTable.this::task);
+    }
+
+    MetadataLogScan(TableOperations ops, Table table, TableScanContext context) {
+      super(ops, table, METADATA_LOG_SCHEMA, MetadataTableType.METADATA_LOG, MetadataLogTable.this::task, context);
+    }
+
+    @Override
+    protected TableScan newRefinedScan(TableOperations ops, Table table, Schema schema, TableScanContext context) {
+      return new MetadataLogScan(ops, table, context);
+    }
+
+    @Override
+    public CloseableIterable<FileScanTask> planFiles() {
+      return CloseableIterable.withNoopClose(MetadataLogTable.this.task(this));
+    }
+  }
+
+  private static StaticDataTask.Row metadataLogTableToRow(TableMetadata.MetadataLogEntry metadataLogEntry,
+                                                          Table table) {
+    Long latestSnapshotId = null;
```

Comment:
Also, very small nit if we change, the null could be eliminated here and next line

Reply:
on removing the null initialization from here, my ide started complaining that `latestSnapshotId` might not have initialized. will revisit.

Reply:
Ah ok, never mind then.